jenkins/opensearch_dashboards/Jenkinsfile

```diff
@@ -0,0 +1,143 @@
+                            // It use executes which does not source the docker container internal ENV VAR
+                            args '-e JAVA_HOME=/usr/lib/jvm/adoptopenjdk-14-hotspot'
+                            alwaysPull true
+                        }
+                    }
+                    steps {
+                        script {
+                            build()
+                        }
+                    }
+                    post() {
+                        always {
+                            cleanWs disableDeferredWipeout: true, deleteDirs: true
+                        }
+                    }
+                }
+                stage('build-arm64') {
+                    agent {
+                        docker {
+                            label 'Jenkins-Agent-al2-arm64-c6g4xlarge-Docker-Host'
+                            image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdk14-node10.24.1-cypress6.9.1-20211005'
+                            // Unlike freestyle docker, pipeline docker does not login to the container and run commands
+                            // It use executes which does not source the docker container internal ENV VAR
+                            args '-e JAVA_HOME=/usr/lib/jvm/adoptopenjdk-14-hotspot'
+                            alwaysPull true
+                        }
+                    }
+                    steps {
+                        script {
+                            build()
+                        }
+                    }
+                    post() {
+                        always {
+                            cleanWs disableDeferredWipeout: true, deleteDirs: true
+                        }
+                    }
+                }
+            }
+            post() {
+                success {
+                    node('Jenkins-Agent-al2-x64-c54xlarge-Docker-Host') {
+                        publishNotification(":white_check_mark:", "Successful Build", "\n${getAllJenkinsMessages()}")
+                    }
+                }
+                failure {
+                    node('Jenkins-Agent-al2-x64-c54xlarge-Docker-Host') {
+                        publishNotification(":warning:", "Failed Build", "")
+                    }
+                }
+            }
+        }
+    }
+}
+
+void build() {
+    git url: 'https://github.com/opensearch-project/opensearch-build.git', branch: 'main'
+
+    sh "./build.sh manifests/$INPUT_MANIFEST"
+    sh './assemble.sh artifacts/manifest.yml'
+
+    script { manifest = readYaml(file: 'artifacts/manifest.yml') }
+    def artifactPath = "${manifest.build.version}/${OPENSEARCH_BUILD_ID}/${manifest.build.architecture}";
+
+    withAWS(role: 'opensearch-bundle', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
+        s3Upload(file: 'artifacts', bucket: "${ARTIFACT_BUCKET_NAME}", path: "builds/dashboards/${artifactPath}")
```

Comment:
is there any intermediate output for dashboards in builds that's useful? In opensearch this is artifacts being published to maven for example; so do we need to upload it?

Reply:
We can build the "snapshots" which would does not include the --release flag to the build [here](https://github.com/opensearch-project/opensearch-build/blob/main/scripts/components/OpenSearch-Dashboards/build.sh#L79). Should publish that as well to the S3 bucket?